lib/FusionInventory/Agent/Task/Inventory/Generic/Dmidecode/Psu.pm

```diff
@@ -0,0 +1,40 @@
+package FusionInventory::Agent::Task::Inventory::Generic::Dmidecode::Psu;
+
+use strict;
+use warnings;
+
+use FusionInventory::Agent::Tools;
+use FusionInventory::Agent::Tools::Generic;
+
+sub isEnabled {
+    my (%params) = @_;
+    return 0 if $params{no_category}->{psu};
+    return 1;
+}
+
+sub doInventory {
+    my (%params) = @_;
+
+    my $inventory = $params{inventory};
+    my $logger    = $params{logger};
+
+    my $infos = getDmidecodeInfos(logger => $logger);
+
+    return unless $infos->{39};
+
+    foreach my $info (@{$infos->{39}}) {
+        my $psu = {
+            PARTNUM => $info->{'Model Part Number'},
+            SERIAL  => $info->{'Serial Number'},
+            VENDOR  => $info->{'Manufacturer'},
+            IS_ATX  => 0,
```

Comment:
The agent don't have to assume the power supply is not ATX if it can find the information and this information is definitively not part of SMBIOS specs. This problem should better be handled on the server-side.